panel/widgets/select.py

```diff
@@ -654,6 +673,73 @@ def _update_options_programmatically(self):
                 self.value = original_values
                 raise
 
+    @classmethod
+    def create_options_from_dataframe(cls, data: pd.DataFrame, columns: List|None=None)->Dict|List:
+        """Returns the options to use with a NestedSelect
+
+        Args:
+            data (pd.DataFrame): A DataFrame to create options from.
+            columns (List | None, optional): The columns to create options from. If None then all
+                columns are used. Defaults to None.
+
+        Returns:
+            Dict|List: A dictionary (or list) of options. The options are sorted ascending.
+        """
+        if columns is None:
+            columns = list(data.columns)
+
+        data = data[columns].drop_duplicates().sort_values(columns).reset_index(drop=True)
+        options = _build_nested_dict(data)
+        return options
+
+    @classmethod
+    def create_from_dataframe(cls, data: pd.DataFrame, columns: List|None=None, **params)->'NestedSelect':
+        """Returns a NestedSelect with options from the data and columns specified
+
+        Args:
+            data (pd.DataFrame): A DataFrame to create options from.
+            columns (List | None, optional): The columns to create options from. If None then all
+                columns are used. Defaults to None. If level is not provided the columns are used as
+                levels.
+            params: Any parameters (besides options) that the NestedSelect accepts
+
+        Returns:
+            NestedSelect: A NestedSelect with options from the data and columns specified
+        """
+        if not columns:
+            columns = list(data.columns)
+
+        options = cls.create_options_from_dataframe(data, columns)
+        params["levels"]=params.get("levels", columns)
+        return NestedSelect(options=options, **params)
+
+    @classmethod
+    def filter_dataframe(cls, data: pd.DataFrame, value: Dict, columns: List|None=None, **params)->pd.DataFrame:
```

Comment:
Should we make this instance method instead? And use `self.value` instead of the provided `value`?

It would make it a bit easier to use but generally less useful.

Reply:
I don't follow why we need this; I don't think NestedSelect should have a method for this. If this functionality is added, it should be under the class method's keyword args (e.g. `subset` kwarg)

Reply:
Agree with @ahuang11 here, I do not think `filter_dataframe` belongs here. There's at least 20 components that would have equal claim to implementing such a method. I get that it's useful, but we can't provide utilities for everything that's useful.

Reply:
Please reconsider. You have `groupby` in hvplot for a reason. This is there for the same reason. Being able to easily create nested selections from a dataset. This is really hard to do correctly and takes time to implement. But its a very common use case.